src/transformers/modeling_utils.py

```diff
@@ -800,6 +821,19 @@ def generate(
                 "Please use another model class (e.g. `OpenAIGPTLMHeadModel`, `XLNetLMHeadModel`, `GPT2LMHeadModel`, `CTRLLMHeadModel`, `T5WithLMHeadModel`, `TransfoXLLMHeadModel`, `XLMWithLMHeadModel`, `BartForConditionalGeneration` )"
             )
 
+        if self.config.is_encoder_decoder:
+            bos_token_id = decoder_start_token_id if decoder_start_token_id is not None else bos_token_id
+            bos_token_id = bos_token_id if bos_token_id is not None else self.config.decoder_start_token_id
+            bos_token_id = bos_token_id if bos_token_id is not None else self.config.bos_token_id
```

Comment:
How did you come up with this hierarchy?

Reply:
Trying not to break the current examples, but I'm also not sure why we should have both decoder_start_token_id and bos_token_id. I don't think they were used together anywhere in the previous code.